engine/src/image.rs

```diff
@@ -40,24 +70,59 @@ impl SvgTangibleObject for Image {
         svg.set_attr("x", position.0.to_string());
         svg.set_attr("y", position.1.to_string());
 
-        let mut defs = Element::new("defs");
-        let mut clip_path = Element::new("clipPath");
-        clip_path.set_attr("id", "clip");
-
-        let mut rect = Element::new("rect");
-        rect.set_attr("width", "100%");
-        rect.set_attr("height", "100%");
-        rect.set_attr("rx", self.round.unwrap_or(0).to_string());
-
-        clip_path.append_child(rect);
+        let padding = self.get_padding();
+
+        if self.round.is_some() || self.shadow.is_some() {
+            let mut defs = Element::new("defs");
+
+            if let Some(round) = self.round {
+                let mut clip_path = Element::new("clipPath");
+                clip_path.set_attr("id", "clip");
+                let mut rect = Element::new("rect");
+                rect.set_attr("width", (size.0 - padding.0 as u32 * 2).to_string());
+                rect.set_attr("height", (size.1 - padding.1 as u32 * 2).to_string());
+                rect.set_attr("x", padding.0.to_string());
+                rect.set_attr("y", padding.1.to_string());
+                rect.set_attr("rx", round.to_string());
+                clip_path.append_child(rect);
+                defs.append_child(clip_path);
+            }
+
+            if let Some((offset, std_deviation)) = self.shadow {
+                let mut filter = Element::new("filter");
+                filter.set_attr("id", "shadow");
+                let mut drop_shadow = Element::new("feDropShadow");
+                drop_shadow.set_attr("dx", offset.0.to_string());
+                drop_shadow.set_attr("dy", offset.1.to_string());
+                drop_shadow.set_attr("stdDeviation", std_deviation.to_string());
+                drop_shadow.set_attr("flood-opacity", "0.6");
```

Comment:
I created a new task to support customizing this and `flood-color`, it’s now black by default https://linear.app/footlights/issue/FL-28

Reply:
We'll commence this issue when the structure is more stable.